Foundation/NSStream.swift

```diff
@@ -202,43 +337,62 @@ public class NSOutputStream : Stream {
     }
     
     public override var streamStatus: Status {
-        return Stream.Status(rawValue: UInt(CFWriteStreamGetStatus(_stream)))!
+        return NSStream.Status(rawValue: UInt(CFWriteStreamGetStatus(_stream)))!
     }
     
     public class func outputStreamToMemory() -> Self {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
+    public func copy(with zone: NSZone?) -> AnyObject{
+        if let url = _url , let shouldAppend = _append {
```

Comment:
not sure if i am handling the copy and mutableCopy responsibilities correctly could not figure out a way with out caching url and shouldAppend and the other instance variables which leads to extra storage on the heap